src/browser/BrowserService.cpp

```diff
@@ -779,6 +781,20 @@ bool BrowserService::removeFirstDomain(QString& hostname)
     return false;
 }
 
+// Gets the base domain of URL, e.g. https://another.example.co.uk -> example.co.uk
+QString BrowserService::baseDomain(const QString& url) const
+{
+    QUrl qurl = QUrl::fromUserInput(url);
+    QString hostname = qurl.host();
+
+    if (hostname.isEmpty() || hostname.indexOf(qurl.topLevelDomain()) < 0) {
+        return {};
+    }
+
+    hostname.chop(qurl.topLevelDomain().length());
+    return hostname.split('.').last().append(qurl.topLevelDomain());
```

Comment:
These two lines need to be broken up and comments added. Took me a while to figure out what was going on.